README.md

```diff
@@ -1,14 +1,59 @@
-# cast-highlight-docker
-
-Docker image packaging for CAST Highlight Automated Code Scan
+# Tidal Source Code Analyzer
+
+When doing source code analysis, Tidal Tools works together with the [CAST Highlight CLI](https://www.castsoftware.com/products/highlight) to execute the analysis.
+
+## High level breakdown
+
+### Online
+
+1. Tidal Tools
+    - Exposes the source code analysis command `tidal analyze code --app-id <your-app-id>` (run from directory containing the source code)
+    - Handles authentication
+    - Hits your Tidal Migrations Workspace `GET apps/#id/cast`
+2. Tidal Migrations Platform
+    - Creates an app in CAST if it's not already present. This CAST app is now associated with your TMP app at the ID provided.
+    - Returns the CAST ID of the app which was created OR the CAST ID of the existing CAST app if you're analyzing the same app again (premium users only).
+3. Tidal Tools
+    - Pulls docker image containing CAST Highlight CLI (that's why you're here)
+    - Runs the CAST CLI in this container
+4. CAST CLI
+    - Analyzes your app and uploads it to CAST with the CAST ID which TMP returned in step 2
+5. Tidal Tools
+    - After the analysis is completed successfully, Tidal Tools will notify TMP that the app has been analyzed
+6. Tidal Migrations Platform
+    - Starts a worker to check on the status of the analysis in CAST. This runs async and periodically checks to see if the analysis is complete
+    - When the analysis is complete, TMP gets the analysis data from CAST and updates your TMP app with the analysis information. This propagates to the frontend and the user is now able to see their source code analysis result when they view their app
+
+![image](https://user-images.githubusercontent.com/43866616/167871482-c18c3fc3-86dd-49d2-a0e9-178a92a7757c.png)
+
+### Offline
+
+1. Tidal Tools
+    - Exposes the source code analysis command `tidal analyze code --app-id <your-app-id>` (run from directory containing the source code). Providing the `--app-id` here is optional. If provided, the ID is saved as a comment with the ZIP file.
```

Comment:
@SamDesmondKing Q do we need to give `--directory` flag for offline analysis?

Reply:
It's optional, same as the online approach. If you don't provide it then it uses the current directory. I've intentionally left it out as this is just a high level overview, I've outlined the `--app-id` flag because this is relevant to our understanding of the feature.